### DISCUSSION TITLE
Run action when pull-request was created OR a new commit was pushed in this PR

### DISCUSSION THREAD

**Discussion by ElenaShlykova** [community/community#25349] [Actions]
<p>I need to run unit tests when the pull-request was created OR a new commit was pushed in this PR.<br>
If the tests fail, the merge should be blocked.<br>
I tried to use <code>on: [push, pull_request]</code>, but it works even when the pull request was not created</p>

**Comment by ghost** [comment, marked as answer]
<p><code>pull_request</code> can be filtered to specific activity types:</p>
<p><a href="https://help.github.com/en/actions/reference/events-that-trigger-workflows#pull-request-event-pull_request" class="onebox" target="_blank">https://help.github.com/en/actions/reference/events-that-trigger-workflows#pull-request-event-pull_request</a></p>
<p>Could you remove <code>push</code> and try using <code>pull_request</code> with the activity type of <code>[created, synchronized]</code> to see if that works?</p>

**Comment by ElenaShlykova** [comment]
<p>Thanks! It works perfectly</p>
